validator/validator.go

```diff
@@ -16,10 +17,28 @@ var (
 	v    *validator.Validate
 )
 
+func validateResourceBounds(fl validator.FieldLevel) bool {
+	req, ok := fl.Parent().Interface().(rpcv8.Transaction)
+	if !ok {
+		return false
+	}
+
+	if (req.Version.String() == "0x3" || req.Version.String() == "0x100000000000000000000000000000003") &&
```

Comment:
I think we can avoid calling the method multiple times
```suggestion
	version := req.Version.String()
	if (version == "0x3" || version == "0x100000000000000000000000000000003") &&
```